### DISCUSSION TITLE
I can not pass step 2

### DISCUSSION THREAD

**Discussion by jaimeslr** [timdorr/tesla-api#318] [Q&A]
I'm always getting this URL when redirected:

https://auth.tesla.com/void/callback?error=login_cancelled&error_description=User%20cancelled%20login&state=123

Any help would be appreciated.

**Comment by HarmOtten** [comment]
I also could not get passed step 2.
But the reason can be different.

@jaimeslr In your case it looks like the api thinks you want to cancel. So maybe you have to look if the hidden field 'cancel' you return in step 2 is empty.
What statusCode to you get in the response of step 2?

In my case I always get statusCode=200, which should be 302, if I follow the desciption.
In step 2 my request looks like:
`headers:{
"Content-Type":"application/x-www-form-urlencoded",
"Cookie":"s%3AvFsQ6-zkN4FyQt79l9Vk0dYsfXi6kdPC.3LXum2Ww9JaRnY6PVxsLxIs1SwrErdGP0dFHdFZpyYs"
}`

`POST  https://auth.tesla.com/oauth2/v3/authorize?client_id=ownerapi&code_challenge=MjA1MmU3NmYwMjdjMDU1MjU1ZDViMjE0OWU5NjRkODU5NDMwOGUyMDkyZGIzZmYwZWM3NGNjYzhlODJiOTYyMg==&code_challenge_method=S256&redirect_uri=https://auth.tesla.com/void/callback&response_type=code&scope=openid email offline_access&state=ABCD`

`body:{
"_csrf":"c419JCwU-nliZTZ0q2xKdYZao0LlBLVXm2OY",
"_phase":"authenticate",
"_process":"1",
"transaction_id":"ilrLtLMl",
"cancel":"",
"identity":"blabla@blablabla.com",
"credential":"********************"
}`

What am I doing wrong?

Updates:
17-FEB-2021: I discovered that I had to disable followRedirects. Now I get statusCode=302 when I test it in Postman. Remains: translating to Node-RED.
18-FEB-2021: I also had to urlencode the scope and the emailaddress. And instead of just the value of the cookie, the "tesla-auth.sid="-part was also necessary. Problem solved!

**Reply by IMgoRt** [reply]
Have you seen the updates in teslaJS:
https://github.com/mseminatore/TeslaJS/commit/e95e896e21132d8921b8d2e435099227d1562310

**Comment by jaimeslr** [comment]
Yeah! That was the problem! Many thanks!
